packages/protocol/contracts/common/FeeHandler.sol

```diff
@@ -61,39 +89,153 @@ contract FeeHandler is Ownable, Initializable, UsingRegistry, ICeloVersionedCont
-        _setRouter(tokens[i], newRouters[i]);
-      }
     }
   }
 
+  function getTokenHandler(address tokenAddress) external view returns (address) {
+    return tokenStates[tokenAddress].handler;
+  }
+
+  function getTokenActive(address tokenAddress) external view returns (bool) {
+    return tokenStates[tokenAddress].active;
+  }
+
+  function getTokenMaxSlippage(address tokenAddress) external view returns (uint256) {
+    return FixidityLib.unwrap(tokenStates[tokenAddress].maxSlippage);
+  }
+
+  function getTokenDailyBurnLimit(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].dailyBurnLimit;
+  }
+
+  function getTokenCurrentDayLimit(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].currentDayLimit;
+  }
+
+  function getTokenToDistribute(address tokenAddress) external view returns (uint256) {
+    return tokenStates[tokenAddress].toDistribute;
+  }
+
+  function setFeeBeneficiary(address beneficiary) external onlyOwner {
+    return _setFeeBeneficiary(beneficiary);
+  }
+
+  function _setFeeBeneficiary(address beneficiary) private {
+    feeBeneficiary = beneficiary;
+    emit FeeBeneficiarySet(beneficiary);
+  }
+
+  function _setBurnFraction(uint256 newFraction) private {
+    FixidityLib.Fraction memory fraction = FixidityLib.wrap(newFraction);
+    require(
+      FixidityLib.lte(fraction, FixidityLib.fixed1()),
+      "Burn fraction must be less than or equal to 1"
+    );
+    burnFraction = fraction;
+    emit BurnFractionSet(newFraction);
+  }
+
+  function setBurnFraction(uint256 fraction) external onlyOwner {
+    return _setBurnFraction(fraction);
+  }
+
+  function sell(address tokenAddress) external {
+    return _sell(tokenAddress);
+  }
+
+  function _addToken(address tokenAddress, address handlerAddress) private {
+    IFeeHandlerSeller(handlerAddress);
+
+    // Check that the contract implements the interface
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    tokenState.active = true;
+    tokenState.handler = handlerAddress;
+
+    activeTokens.push(tokenAddress);
+  }
+
+  function addToken(address tokenAddress, address handlerAddress) external onlyOwner {
+    _addToken(tokenAddress, handlerAddress);
+  }
+
+  function getActiveTokens() public view returns (address[] memory) {
+    return activeTokens;
+  }
+
+  function removeToken(address tokenAddress) external {}
+
+  function _sell(address tokenAddress) private onlyWhenNotFrozen nonReentrant {
+    IERC20 token = IERC20(tokenAddress);
+
+    TokenState storage tokenState = tokenStates[tokenAddress];
+    FixidityLib.Fraction memory balanceOfTokenToBurn = FixidityLib.newFixed(
+      token.balanceOf(address(this)).sub(tokenState.toDistribute)
+    );
+
+    uint256 balanceToBurn = (burnFraction.multiply(balanceOfTokenToBurn).fromFixed());
+
+    tokenState.toDistribute += (token.balanceOf(address(this)).sub(balanceToBurn));
```

Comment:
If the contract doesnt have enough of balance to burn it will revert. We could consider burning what it has and just subtract it

Reply:
I'll add a min(balanceToBurn, balanceInContract), but we should never enter that situation, because the contract should always have balanceToBurn + toDistribute.